src/main/java/graphql/execution/instrumentation/Instrumentation.java

```diff
@@ -230,6 +232,20 @@ default ExecutionStrategyInstrumentationContext beginExecutionStrategy(Instrumen
         return beginExecutionStrategy(parameters.withNewState(state));
     }
 
+    /**
+     * This is called each time an {@link graphql.execution.ExecutionStrategy} object resolution is called, which may be multiple times
+     * per query as the engine recursively descends over the query.
+     *
+     * @param parameters the parameters to this step
+     * @param state      the state created during the call to {@link #createState(InstrumentationCreateStateParameters)}
+     *
+     * @return a nullable {@link ExecutionStrategyInstrumentationContext} object that will be called back when the step ends (assuming it's not null)
+     */
+    @Nullable
+    default ExecuteObjectInstrumentationContext beginExecuteObject(InstrumentationExecutionStrategyParameters parameters, InstrumentationState state) {
+        return ExecuteObjectInstrumentationContext.NOOP;
+    }
+
```

Comment:
This is the Instrumentation callback for when we recursively call back to the execution strategy during "completeObject()"

Before we used to call `beginExecutionStrategy` but thats not really semantically correct any more.  We could call this old `beginExecutionStrategy` but its kinda wrong